fuel-block-producer/src/block_producer/transaction_selector.rs

```diff
@@ -0,0 +1,173 @@
+use std::{
+    cmp::Reverse,
+    ops::Deref,
+    sync::Arc,
+};
+
+use fuel_core_interfaces::common::{
+    fuel_tx::CheckedTransaction,
+    fuel_types::Word,
+};
+
+pub fn select_transactions(
+    mut includable_txs: Vec<Arc<CheckedTransaction>>,
+    max_gas: Word,
+) -> Vec<CheckedTransaction> {
+    // Select all txs that fit into the block, preferring ones with higher gas price.
+    //
+    // Future improvements to this algorithm may take into account the parallel nature of
+    // transactions to maximize throughput.
+    let mut used_block_space: Word = 0;
+
+    // Sort transactions by gas price, highest first
+    includable_txs.sort_by_key(|a| Reverse(a.transaction().gas_price()));
+
+    // Pick as many transactions as we can fit into the block (greedy)
```

Comment:
Should we have some method to seperate `select_transaction` strategies? So either renaming it, or using some shared interface could be helpful. So that block producers can choose between `select_tx_gas_sorted` and `select_mev_tx_priority` etc, etc

Reply:
This is outside the scope of this PR. Maybe you could open a new issue if that needs discussion?